lib/kvdb/kvdb_ctxn.c

```diff
@@ -108,6 +126,7 @@ kvdb_ctxn_bind_putref(struct kvdb_ctxn_bind *bind)
 
     if (kvdb_ctxn_trylock_impl(kvdb_ctxn_h2r(bind->b_ctxn)) != 0)
         return;
+    // How can a putref be ignored if txn is not active?
```

Comment:
This one worries me.  This kind of logic tells me it's not really a ref count.  With most ref count logic: 1) the caller knows for sure if it has a ref, and 2) when a ref needs to be put, is must be put or you have a leak.